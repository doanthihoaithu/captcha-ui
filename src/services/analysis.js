import { requestServices } from 'services';
import * as constants from 'constants/index'


const analyzeQuery = (accessToken, data) => requestServices.customAxios.post(
    constants.analysis_analyze_query,
    data,
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data'
        }
    }
)

const generateQuery = (data) => requestServices.customAxios.post(
    constants.generate_captcha_query,
    data,
    {
        headers: {
            // 'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data',
            // 'Content-Type': 'application/json'
        }
    }
)

export default {
    analyzeQuery,
    generateQuery
}