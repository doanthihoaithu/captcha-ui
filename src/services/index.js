// @flow
export { default as userServices } from './users';
export { default as requestServices } from './requests';
export { default as rootReducer } from './rootReducer';
export { default as store } from './store'
