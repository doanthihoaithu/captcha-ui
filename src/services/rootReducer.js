import { combineReducers } from 'redux';
import userReducer from './redux/user/reducers';
import statusReducer from './redux/status/reducers';

export default combineReducers({
	user: userReducer,
	status: statusReducer
});