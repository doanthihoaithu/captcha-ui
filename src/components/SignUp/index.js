import animationTrigger from 'helpers/loadingAnimationTrigger';
import React, { useRef, useState } from 'react';
import { userServices } from 'services';
import { Link } from 'react-router-dom'

function SignUp(props) {
  const username = useRef(null);
  const email = useRef(null);
  const password = useRef(null);
  const fullname = useRef(null);

  const [registered, changeRegistrationStatus] = useState(false);
  const [somethingWronged, changeSomethingWrongedStatus] = useState(false);

  function registerNewUser(username, email, password, fullname) {
    animationTrigger.startLoading();
    let data = { username, email, password, fullname };
    // console.log(_username, _email, _password)
    userServices.registerNewUser(data).then((res) => {
      console.log(res);

      changeRegistrationStatus(true);
      changeSomethingWrongedStatus(false);
      animationTrigger.stopLoading()
    }).catch(err => {
      console.log(err)
      changeSomethingWrongedStatus(true);
      animationTrigger.stopLoading()
    })
  }

  function onFormSubmit(event) {
    event.preventDefault();

    const valid = event.target.checkValidity();
    // console.log(valid)
    if (valid) {
      registerNewUser(username.current.value, email.current.value, password.current.value, fullname.current.value);
    }
  }

  return (
    <div className="container-contact100 animated fadeIn fast ">
      <div className="wrap-contact100">
        <button className="btn-hide-contact100">
          <i className="zmdi zmdi-close"></i>
        </button>

        <div className="contact100-form-title" style={{ backgroundImage: `url(${require('./../../asset/images/bg-02.jpg')})` }}>
          <span>Create new account</span>
        </div>

        {
          registered ? <SuccessfulRegistration /> :
            <form className="contact100-form validate-form" onSubmit={onFormSubmit}>
              <div className="wrap-input100 validate-input">
                <input pattern="^(?=.{6,150}$)[a-zA-Z0-9]+$" type="text" ref={username} className="input100" id="exampleInputUsername1" aria-describedby="usernameHelp" placeholder="Enter username" required />
                <span className="focus-input100"></span>
                <label className="label-input100" htmlFor="username">
                  <span className="lnr lnr-user m-b-2"></span>
                </label>
              </div>

              <div className="wrap-input100 validate-input">
                <input id="name" ref={fullname} className="input100" type="text" placeholder="Full name" required />
                <span className="focus-input100"></span>
                <label className="label-input100" htmlFor="name">
                  <span className="lnr lnr-user m-b-2"></span>
                </label>
              </div>

              <div className="wrap-input100 validate-input">
                <input type="email" ref={email} className="input100" id="exampleEmail1" placeholder="Email" required />
                <span className="focus-input100"></span>
                <label className="label-input100" htmlFor="email">
                  <span className="lnr lnr-envelope m-b-5"></span>
                </label>
              </div>

              <div className="wrap-input100 validate-input">
                <input pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$" type="password" ref={password} className="input100" id="exampleInputPassword1" placeholder="Password" required />
                <span className="focus-input100"></span>
                <label className="label-input100" htmlFor="password">
                  <span className="lnr lnr-lock m-b-5"></span>
                </label>
              </div>

              <div className="w-100 text-center mb-1"><span className="text-danger" hidden={!somethingWronged}>Something went wrong. Please try again!</span></div>

              <div className="container-contact100-form-btn">
                <button className="contact100-form-btn" type="submit">
                  Create account
            </button>
              </div>
            </form>
        }

        <div className="d-flex flex-column justify-content-center align-items-center mb-5">
          <div className="">
            <small>Already have an account? </small><Link to="/signin" className="text-info">Sign In</Link>
          </div>
        </div>
      </div>
    </div>
  )
};

function SuccessfulRegistration() {
  return (
    <div className="d-flex align-items-center justify-content-center m-5 flex-column">
      <h4 className="text-success text-center">You have successfully registered </h4>
    </div>
  )
}

export default SignUp;