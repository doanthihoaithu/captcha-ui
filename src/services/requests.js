// @flow
import axios from 'axios';
import * as actions from 'services/redux/user/actions';
import { store } from 'services'
import { toast } from 'react-toastify'

export const BASE_API_URL: string = `${process.env.REACT_APP_BASE_API_URL || 'localhost:3000'}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    // console.log(error.response.data)
    if (error.response && error.response.data.code === 500) {
      switch (error.response.data.message) {
        case "Signature has expired":
          store.dispatch(actions.logOut());
          toast.error('Your session has expired. Please sign in to continue');
          return Promise.reject(error);
        default:
          return Promise.reject(error);
      }
    }

    return Promise.reject(error);
  },
);

export default { customAxios };
