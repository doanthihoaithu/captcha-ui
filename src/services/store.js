import { createStore } from 'redux';
import { rootReducer } from './../services';

const store = createStore(rootReducer);

export default store;
