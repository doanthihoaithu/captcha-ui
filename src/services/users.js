// @flow
import { requestServices } from 'services';
import * as constants from 'constants/index'

// const fetchUsers = () => requestServices.customAxios.get('/').then(res => res.data);

const signUserIn = (data) => requestServices.customAxios.post(
    constants.user_authenticate_address,
    data,
    {
        header: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*',
        }
    }
)

const signUserOut = (accessToken) => requestServices.customAxios.post(
    constants.user_sign_out_address,
    {},
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const registerNewUser = (data) => requestServices.customAxios.post(
    constants.user_sign_up_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const resetPassword = (data) => requestServices.customAxios.post(
    constants.user_forget_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const changePassword = (data, accessToken) => requestServices.customAxios.post(
    constants.user_change_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${accessToken}`
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const updateAvatar = (username, data, accessToken) => requestServices.customAxios.post(
    constants.user_update_avatar_address + username,
    data,
    {
        headers: {
            'Content-Type': 'multipart/form-data',
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const getAvatar = (username) => requestServices.customAxios.get(
    constants.user_get_avatar_address + username,
)

export default {
    signUserIn,
    signUserOut,
    registerNewUser,
    resetPassword,
    changePassword,
    updateAvatar,
    getAvatar,
};
