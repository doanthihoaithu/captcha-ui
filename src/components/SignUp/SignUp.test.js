import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import SignUp from '.';
import { store, userServices } from './../../services';


let component;
let registerNewUserAPI;


beforeEach(() => {
    component = mount(
        <Router>
            <Provider store={store}>
                <SignUp />
            </Provider>
        </Router>
    )
    registerNewUserAPI = jest.spyOn(userServices, 'registerNewUser').mockReturnValue(Promise.resolve("value"));
})

afterEach(() => {
    registerNewUserAPI.mockReset();
    component.unmount();
})

it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
})


it('should not call function registerNewUser when submit nothing', () => {
    let registerNewUserForm = component.find("form");

    registerNewUserForm.simulate('submit');

    registerNewUserAPI.mockClear();
    expect(registerNewUserAPI).toHaveBeenCalledTimes(0);
})

it('should call function registerNewUser when submit valid data', async () => {
    let registerNewUserForm = component.find("form");

    component.find("input[id='exampleInputUsername1']").getDOMNode().value = ("username")
    component.find("input[id='name']").getDOMNode().value = ("User Full Name")
    component.find("input[id='exampleEmail1']").getDOMNode().value = ("useremail@example.com")
    component.find("input[id='exampleInputPassword1']").getDOMNode().value = ("newpassword")

    component.update()

    expect(component.find("input")).toHaveLength(4)
    await registerNewUserForm.simulate('submit');

    expect(registerNewUserAPI).toHaveBeenCalledTimes(1);
})

it('should not call function registerNewUser when submit invalid data', async () => {
    let registerNewUserForm = component.find("form");

    component.find("input[id='exampleInputUsername1']").getDOMNode().value = ("username")
    component.find("input[id='name']").getDOMNode().value = ("User Full Name")
    component.find("input[id='exampleEmail1']").getDOMNode().value = ("wrongEmail")
    component.find("input[id='exampleInputPassword1']").getDOMNode().value = ("newpassword")

    component.update()

    expect(component.find("input")).toHaveLength(4)
    await registerNewUserForm.simulate('submit');

    expect(registerNewUserAPI).toHaveBeenCalledTimes(0);
})