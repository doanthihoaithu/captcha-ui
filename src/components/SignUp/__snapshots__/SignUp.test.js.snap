// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <SignUp>
        <div
          className="container-contact100 animated fadeIn fast "
        >
          <div
            className="wrap-contact100"
          >
            <button
              className="btn-hide-contact100"
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
            <div
              className="contact100-form-title"
              style={
                Object {
                  "backgroundImage": "url(bg-02.jpg)",
                }
              }
            >
              <span>
                Create new account
              </span>
            </div>
            <form
              className="contact100-form validate-form"
              onSubmit={[Function]}
            >
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  aria-describedby="usernameHelp"
                  className="input100"
                  id="exampleInputUsername1"
                  pattern="^(?=.{6,150}$)[a-zA-Z0-9]+$"
                  placeholder="Enter username"
                  required={true}
                  type="text"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="username"
                >
                  <span
                    className="lnr lnr-user m-b-2"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="name"
                  placeholder="Full name"
                  required={true}
                  type="text"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="name"
                >
                  <span
                    className="lnr lnr-user m-b-2"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="exampleEmail1"
                  placeholder="Email"
                  required={true}
                  type="email"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="email"
                >
                  <span
                    className="lnr lnr-envelope m-b-5"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="exampleInputPassword1"
                  pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$"
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="password"
                >
                  <span
                    className="lnr lnr-lock m-b-5"
                  />
                </label>
              </div>
              <div
                className="w-100 text-center mb-1"
              >
                <span
                  className="text-danger"
                  hidden={true}
                >
                  Something went wrong. Please try again!
                </span>
              </div>
              <div
                className="container-contact100-form-btn"
              >
                <button
                  className="contact100-form-btn"
                  type="submit"
                >
                  Create account
                </button>
              </div>
            </form>
            <div
              className="d-flex flex-column justify-content-center align-items-center mb-5"
            >
              <div
                className=""
              >
                <small>
                  Already have an account? 
                </small>
                <Link
                  className="text-info"
                  to="/signin"
                >
                  <a
                    className="text-info"
                    href="/signin"
                    onClick={[Function]}
                  >
                    Sign In
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </SignUp>
    </Provider>
  </Router>
</BrowserRouter>
`;
